### PR TITLE
Enhancement/66 tighten eslint rules

### LINKED ISSUE #66: Discuss and tighten eslint rules
In order for linting to pass for now I've disabled a couple of rules.
```   "max-len": "off",
        "react/prop-types": "off",
        "react/destructuring-assignment": "off",
        "react/no-unescaped-entities": "off",
        "global-require": "off",
        "react/jsx-props-no-spreading": "off",
        "jsx-a11y/label-has-associated-control": "off",
        "prefer-destructuring": "off",
        "jsx-a11y/click-events-have-key-events": "off",
        "jsx-a11y/no-static-element-interactions": "off"
```

We should discuss which ones we want to turn back on and enable them and fix the code

### PR DESCRIPTION
**Pull request by florianschmidt1994**
**What changes does this PR introduce**
- Enable react/jsx-props-no-spreading eslint rule and fix code to be valid
- Enable react/destructuring-assignment eslintrc rule and fix code to be valid
- Enable react/no-unescaped-entities eslint rule  and fix code to be valid
- Remove jsx-a11y/click-events-have-key-events: off from eslintrc
- Enable jsx-a11y/label-has-associated-control eslint rule  and fix code to be valid
- Enable jsx-a11y/no-static-element-interactions eslint rule and fix code to be valid
- Enable prefer-destructuring eslint rule (19 hours ago)  and fix code to be valid

Please check those changes and I would appreciate if you could also test that everything is working as expected. I know some changes might be controversial, we can discuss them here! Each rule is in a separate commit, so we could also cherry-pick the ones that we want to have

This PR closes #66

### CODE REVIEW

File: src/components/LocationInput.jsx [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+  const {
+    required,
+    value,
+    onChange,
+    onSelect,
+  } = props;
+
   if (isMapsApiEnabled) {
-    return <Autocomplete {...props} />;
+    return <Autocomplete required={required} value={value} onChange={onChange} onSelect={onSelect} />;
```
Comment (kenodressel) [inline src/components/LocationInput.jsx:16]: I agree with most of the changes made here, this one seems a little weird, I see the point but I feel like its really tedious. Whats your opinion @jagaSto @Maurice22 @tgraupne ?

File: src/views/FAQ.jsx [hunk 2]
```diff
@@ -6,30 +6,37 @@ import Footer from '../components/Footer';
...
     );
   };
 
-  const QA = (props) => (
-    <>
-      <h2 className="text-xl font-teaser mt-8">{props.question}</h2>
-      <p className="font-open-sans">{props.children}</p>
-    </>
-  );
+  const QA = (props) => {
```
Comment (kenodressel) [inline src/views/FAQ.jsx:27]: could we use argument in place deconstruction?
```suggestion
  const QA = ({ question, children }) => {
```

File: src/views/Impressum.jsx [hunk 3]
```diff
@@ -80,8 +80,8 @@ export default function Impressum() {
             <br />
           </p>
           <p>
-            Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. (''Google'').
-            Google Analytics verwendet sog. ''Cookies'', Textdateien, die auf Ihrem Computer gespeichert werden und die eine Analyse der Benutzung der
+            Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. (&apos;&apos;Google&apos;&apos;).
+            Google Analytics verwendet sog. &apos;&apos;Cookies&apos;&apos;, Textdateien, die auf Ihrem Computer gespeichert werden und die eine Analyse der Benutzung der
```
Comment (kenodressel) [inline src/views/Impressum.jsx:84]:
```suggestion
            Google Analytics verwendet sog. &quot;Cookies&quot;, Textdateien, die auf Ihrem Computer gespeichert werden und die eine Analyse der Benutzung der
```

File: src/views/Impressum.jsx [hunk 4]
```diff
@@ -80,8 +80,8 @@ export default function Impressum() {
             <br />
           </p>
           <p>
-            Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. (''Google'').
-            Google Analytics verwendet sog. ''Cookies'', Textdateien, die auf Ihrem Computer gespeichert werden und die eine Analyse der Benutzung der
+            Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. (&apos;&apos;Google&apos;&apos;).
```
Comment (kenodressel) [inline src/views/Impressum.jsx:83]: I suggest we use double quotes?
```suggestion
            Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. (&quot;Google&quot;).
```

File: src/views/Press.jsx [hunk 5]
```diff
@@ -26,15 +26,34 @@ export default function Press() {
...
-  const Article = (props) => (
-    <div className="bg-kaki p-4 mb-4 mt-4 font-open-sans w-full">
-      <div className="text-xs text-gray-700">{props.date}</div>
-      <div className="font-bold">{props.title}</div>
-      {props.link
-        ? <a className="text-secondary w-full block truncate" href={props.link} target="_blank" rel="nofollow noopener noreferrer">{props.link}</a>
-        : <span className="w-full block truncate">{props.text}</span>}
-    </div>
-  );
+  const Article = (props) => {
```
Comment (kenodressel) [inline src/views/Press.jsx:29]: again inline deconstruction

File: src/components/LocationInput.jsx (hunk 1 above)
Comment (florianschmidt1994) [reply src/components/LocationInput.jsx:16]: I know, felt a little weird to me, too. I added this chance here for the sake of consistency

**Comment by Maurice22** [comment]
There seems to be an issue with the location input introduced by this PR:
Bildschirmfoto 2020-04-16 um 17 41 43

File: src/views/Press.jsx [hunk 6]
```diff
@@ -197,7 +207,15 @@ export default function Press() {
             <img alt="bunte_de" src={require('../assets/bunte_de.jpg')} />
           </div>
         </div>
-        {articles.map((article) => <Article key={article.link || article.text} {...article} />)}
+        {articles.map((article) => (
```
Comment (kenodressel) [inline src/views/Press.jsx:210]: Should we deconstruct here?

File: src/components/LocationInput.jsx [hunk 7]
```diff
@@ -212,12 +220,12 @@ function AutocompleteSuggestion(props) {
   } = props;
 
   return (
-    <div
+    <button
+      type="button"
       className="px-3 py-2 hover:bg-gray-100 cursor-pointer"
```
Comment (kenodressel) [inline src/components/LocationInput.jsx:225]:
```suggestion
      className="px-3 py-2 hover:bg-gray-100 cursor-pointer w-full text-left"
```

Comment (kenodressel) [reply src/components/LocationInput.jsx:225]: should fix the mentioned issue @Maurice22

File: src/components/LocationInput.jsx [hunk 8]
```diff
@@ -1,14 +1,22 @@
...
 import { getSuggestions } from '../services/GeoService';
 
 export default function LocationInput(props) {
+  const {
+    required,
+    value,
+    onChange,
+    onSelect,
+  } = props;
+
```
Comment (Maurice22) [inline src/components/LocationInput.jsx:14]: this is missing the fullText prop

File: src/components/LocationInput.jsx (hunk 7 above)
Comment (Maurice22) [reply src/components/LocationInput.jsx:225]: Working on it

File: src/components/LocationInput.jsx (hunk 8 above)
Comment (Maurice22) [reply src/components/LocationInput.jsx:14]: working on it

File: src/components/entry/Entry.jsx [hunk 9]
```diff
@@ -198,7 +198,8 @@ export default function Entry(props) {
         ) : null}
         {attemptingToReport
           ? (
-            <div
+            <button
+              type="button"
```
Comment (Maurice22) [inline src/components/entry/Entry.jsx:202]: Broken but working on it

Comment (Maurice22) [reply src/components/entry/Entry.jsx]: Fixed

File: src/views/Press.jsx (hunk 6 above)
Comment (Maurice22) [reply src/views/Press.jsx:210]: Wouldn't do so as we also don't do it for the entry.. I also think this is easier to read

**Comment by Maurice22** [comment]
@florianschmidt1994, can you check the failed pipeline? Locally, it does not fail for me and the annotated line does not contain any props

**Comment by florianschmidt1994** [comment]
@Maurice22 I've rerun the tests and they pass now. Seems to be a bug / race condition with the github action

**Pull request merged**